addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1044,6 +1044,21 @@ options.registry.ReplaceMedia.include({
         }
         return this._super(...arguments);
     },
+    /**
+     * @override
+     */
+    _computeVisibility() {
+        // `o_editable_media` class forces the option to be displayed even if
+        // the target matches the data-exclude (see `computeSelectorFunctions`).
+        // For social media linked to a DB record, we don't want to display the
+        // Replace Media button and URL field, but we do want to enable the icon
+        // to be changed on double-click (this remains a hidden feature).
+        if (this.$target[0].matches("a[href^='/website/social/'] > i.fa")) {
+            return false;
+        }
+        return this._super(...arguments);
+    },
```

Comment:
I'm not a huge fan of this part.. IMHO the data exclude should suffice even if there's a `o_editable_media` class, but it looks like the code has been expressly written to make the presence of the class more important than the exclude. I wonder if we couldn't do better in `filterFunc` which is used in `_computeSelectorFunctions`. Something like making the exclude part coming from the data-exclude more important than the `o_editable_media` class :thinking:

Reply:
@Guillaume-gdi 

> Something like making the exclude part coming from the data-exclude more important than the o_editable_media class 🤔

Oh yes, indeed, this is an error. data-exclude should be more important than o_editable_media. You should fix that too please :pray: Do that in a different commit, in a different PR since you'll have to merge that further than 15.2.

Although in stable, I'd still recommend to use `_computeVisibility` here (to not have to update the XML), then in master we can switch to using data-exclude (which in the OWL branch will actually be described in JS too).

Reply:
> Although in stable, I'd still recommend to use _computeVisibility here (to not have to update the XML), then in master we can switch to using data-exclude (which in the OWL branch will actually be described in JS too).

I forgot to mention that there is already the data-exclude on the XML option declaration of `ReplaceMedia` so we can do as you prefer:
merge this PR as it is actually (with the `_computeVisibility` AND the data-exclude) and we remove the `_computeVisibility` in the forward port of the other fix (data-exclude > o_editable_media)
or
merge the other fix first (data-exclude > o_editable_media) and wait for its fw port to reach 15.2 and remove the `_computeVisibility` here

The other fix (data-exclude > o_editable_media): https://github.com/odoo/odoo/pull/132849

Reply:
Oh, well, obviously the second solution then. If the data-exclude is already there but not working, fixing it through the other PR is enough.